mdagent/tools/base_tools/analysis_tools/pca_tools.py

```diff
@@ -0,0 +1,370 @@
+        return result
+
+    def _get_number_pcs(self):
+        self._sub_pcs = self._sub_array_sum_to_m(
+            self.pc.explained_variance_ratio_, self.pc_percentage
+        )
+        self.n_pcs = len(self._sub_pcs)
+        if self.n_pcs > 3:
+            self.n_pcs = 3
+        return self.n_pcs
+
+    def _make_transformation(self):
+        self.transformed = self.reduce_dim[:, : self.n_pcs]
+
+    def _make_df(self):
+        self._get_number_pcs()
+        self._make_transformation()
+        self.pc_df = pd.DataFrame(
+            self.transformed, columns=["PC{}".format(i + 1) for i in range(self.n_pcs)]
+        )
+        self.pc_df["Time (ps)"] = self.pc_df.index * self.traj.timestep
+
+    def _cosine_content(self, pca_space, i):
+        """Measure the cosine content of the PCA projection.
+
+        The cosine content of pca projections can be used as an indicator if a
+        simulation is converged. Values close to 1 are an indicator that the
+        simulation isn't converged. For values below 0.7 no statement can be made.
+        If you use this function please cite [BerkHess1]_.
+        References
+        ----------
+        .. [BerkHess1] Berk Hess. Convergence of sampling in protein simulations.
+                    Phys. Rev. E 65, 031910 (2002).
+        """
+        t = np.arange(len(pca_space))
+        T = len(pca_space)
+        cos = np.cos(np.pi * t * (i + 1) / T)
+        return (
+            (2.0 / T)
+            * (scipy.integrate.simps(cos * pca_space[:, i])) ** 2
+            / scipy.integrate.simps(pca_space[:, i] ** 2)
+        )
+
+    def make_scree_plot(self):
+        extra_mess = ""
+        if not self.pc:
+            pc_mess = self.get_pc()
+            if "not aligned" in pc_mess:
+                extra_mess += pc_mess
+        if not self.n_pcs:
+            self._get_number_pcs()
+        cumulative_variance = self.pc.explained_variance_ratio_.cumsum()
+        plt.plot(cumulative_variance)
+        # Calculate the index where cumulative variance exceeds or meets 95%
+        threshold_index = len(self._sub_pcs) - 1
+
+        # Add a horizontal dashed line at 95% threshold
+        plt.axvline(
+            x=threshold_index,
+            color="r",
+            linestyle="--",
+            label=f"{self.pc_percentage*100}% Contribution at PC {threshold_index+1}",
+        )
+
+        plt.xlabel("Principal component")
+        plt.ylabel("EigenValue Proportion")
+        plt.legend()
+
+        desc = f"Scree Plot of the eigenvalues of simulation {self.sim_id}"
+        file_name = self.path_registry.write_file_name(
+            FileType.FIGURE,
+            fig_analysis="scree_plot",
+            file_format="png",
+            Sim_id=self.sim_id,
+        )
+        plot_id = self.path_registry.get_fileid(file_name, FileType.FIGURE)
+        path = self.path_registry.ckpt_dir + "/figures/"
+        plt.savefig(path + file_name)  # Save the figure before plt.show()
```

Comment:
```suggestion
        plt.savefig(path + file_name) 
```

This comment appears several times in this file, can we remove it? especially since we aren't actually showing the plot anywhere

Reply:
oki. Ive noticed this comes because of auto-completion with copilot is the root reason of these repetitive comments haha. I've deal with two of them so far, and if I keep seeing them, ill remove them too :)